components/camel-tahu/src/main/java/org/apache/camel/component/tahu/TahuEndpoint.java

```diff
@@ -0,0 +1,268 @@
+    private final TahuConfiguration configuration;
+
+    @UriParam(label = "producer (edge node only),advanced", description = "Flag enabling support for metric aliases",
+              defaultValue = "false")
+    @Metadata(applicableFor = TahuConstants.EDGE_NODE_SCHEME)
+    private boolean useAliases = false;
+
+    @UriParam(label = "producer,advanced",
+              description = "To use a custom HeaderFilterStrategy to filter headers used as Sparkplug metrics",
+              defaultValueNote = "Defaults to sending all Camel Message headers with name prefixes of \""
+                                 + TahuConstants.METRIC_HEADER_PREFIX + "\", including those with null values")
+    @Metadata(applicableFor = { TahuConstants.EDGE_NODE_SCHEME, TahuConstants.DEVICE_SCHEME })
+    private volatile HeaderFilterStrategy headerFilterStrategy;
+
+    @UriParam(label = "producer (edge node only),advanced",
+              description = "To use a specific org.eclipse.tahu.message.BdSeqManager implementation to manage edge node birth-death sequence numbers",
+              defaultValue = "org.apache.camel.component.tahu.CamelBdSeqManager")
+    @Metadata(applicableFor = TahuConstants.EDGE_NODE_SCHEME)
+    private volatile BdSeqManager bdSeqManager;
+
+    @UriParam(label = "producer (edge node only),advanced",
+              description = "Path for Sparkplug B NBIRTH/NDEATH sequence number persistence files. This path will contain files named as \"<Edge Node ID>-bdSeqNum\" and must be writable by the executing process' user",
+              defaultValue = "${sys:java.io.tmpdir}/CamelTahuTemp")
+    @Metadata(applicableFor = TahuConstants.EDGE_NODE_SCHEME)
+    private String bdSeqNumPath;
+
+    @UriPath(label = "consumer", description = "ID for the host application")
+    @Metadata(applicableFor = TahuConstants.HOST_APP_SCHEME, required = true)
+    private final String hostId;
+
+    TahuEndpoint(String uri, TahuComponent component, TahuConfiguration configuration,
+                 String groupId, String edgeNode) {
+        this(uri, component, configuration, groupId, edgeNode, null);
+    }
+
+    TahuEndpoint(String uri, TahuComponent component, TahuConfiguration configuration,
+                 String groupId, String edgeNode, String deviceId) {
+        super(uri, component);
+
+        this.configuration = configuration;
+
+        this.groupId = groupId;
+        this.edgeNode = edgeNode;
+        this.deviceId = deviceId;
+        this.hostId = null;
+
+    }
+
+    TahuEndpoint(String uri, TahuComponent component, TahuConfiguration configuration, String hostId) {
+        super(uri, component);
+
+        this.configuration = configuration;
+
+        this.groupId = null;
+        this.edgeNode = null;
+        this.deviceId = null;
+        this.hostId = hostId;
+
+    }
+
+    @Override
+    public Producer createProducer() throws Exception {
+        if (ObjectHelper.isEmpty(getGroupId()) || ObjectHelper.isEmpty(getEdgeNode())) {
```

Comment:
Same not as before about `ObjectHelper.isNotNullOrEmpty`.

Reply:
Done.